src/actions/action.surrounder.ts

```diff
@@ -0,0 +1,72 @@
+import { Context } from "telegraf";
+import { Message, User } from "telegraf/typings/core/types/typegram";
+import { AxiosError } from "axios";
+
+export default abstract class Action {
+  id: string;
+  message: Message.CommonMessage;
+  inline_message_id: string;
+  from: User;
+  item_name: string;
+  context: Context;
+
+  get ctx() {
+    return this.context;
+  }
+
+  set ctx (ctx: Context) {
+    this.context = ctx;
+    this.id = (ctx.callbackQuery as any).data.split(':')[1];
+    this.message = ctx.callbackQuery.message;
+    this.inline_message_id = ctx.callbackQuery.inline_message_id;
+    this.from = ctx.callbackQuery.from;
+  }
+
+  async execute(): Promise<void> {
+    await this.updateState();
+
+    const extra: object = {
+      parse_mode: 'HTML',
+    }
+
+    this.addMarkup(extra);
+
+    await this.ctx.telegram.editMessageText(
+        this.message.chat.id,
+        this.message.message_id,
+        this.inline_message_id,
+        this.createMessage(),
+        extra,
+    )
+  }
+
+  async catch(e) {
+    const axiosError = e as AxiosError;
+
+    if (axiosError.isAxiosError) {
+      if (axiosError.response?.status === 404) {
+        await this.ctx.telegram.editMessageText(
+            this.message.chat.id,
+            this.message.message_id,
+            this.inline_message_id,
+            `<b>${this.item_name} ${this.id} вже було видалено.</b>`,
+            {
+              parse_mode: 'HTML',
+            }
```

Comment:
Maybe extract this object to an upper level constant and reuse, because we also use it in the `execute` method.

Reply:
Or move it to `addMarkup`